core/src/main/java/org/kohsuke/stapler/compression/FilterServletOutputStream.java

```diff
@@ -11,9 +14,16 @@
  */
 public class FilterServletOutputStream extends ServletOutputStream {
     private final OutputStream out;
-
-    public FilterServletOutputStream(OutputStream out) {
+    private final ServletOutputStream realSream;
+    
+    /**
+     * Constructs a new {@link FilterOutputStream}.
+     * @param out the stream that sits above the realStream, performing some filtering.  This must be eventually delegating eventual writes to {@code realStream}.
+     * @param realStream the actual underlying ServletOutputStream from the container.  Used to check the {@link #isReady()} state and to add {@link WriteListener}s. 
+     */
+    public FilterServletOutputStream(OutputStream out, ServletOutputStream realStream) {
```

Comment:
this is pretty ugly - there is no need for stapler to do gzip at all - see #130

Reply:
The usage of this in Jenkins core shows that this is not appropriate there.
I found 2 external uses of this 
1 in jenkins core
1 in a propriatary plugin

in both cases it is a mock stream, that does not respond to the client at all and as such has a different use case to this.